ros/bbn_integration/src/nodes/zmq_integration_client.cxx

```diff
@@ -0,0 +1,156 @@
+  : public rclcpp::Node
+{
+private:
+  std::string m_server_address;
+  // ZMQ context with a single IO thread
+  zmq::context_t m_zmq_context{ 1 };
+  // ZMQ socket for communication
+  zmq::socket_t m_zmq_socket{ m_zmq_context, zmq::socket_type::req };
+
+  rclcpp::Subscription< BBNUpdate >::SharedPtr m_sub_updates;
+
+public:
+  /// Constructor
+  ZmqIntegrationClient( rclcpp::NodeOptions const& options )
+    : rclcpp::Node( "ZmqIntegrationClient", options )
+  {
+    this->declare_parameter( "topic_update_msg" );
+    this->declare_parameter( "server_address" );
+
+    // Separating parameter declaration and getting so the error messages are
+    // more (read: at all) informative.
+    auto topic_update_msg = this->get_parameter( "topic_update_msg" ).as_string();
+    m_server_address = this->get_parameter( "server_address" ).as_string();
+
+    // Initialize the socket with the server address
+    RCLCPP_INFO(
+      get_logger(),
+      "Initializing ZMQ socket for the server address: " + m_server_address );
+    if( !std::regex_match( m_server_address, TCP_URL_REGEX() ) )
+    {
+      throw std::invalid_argument( "Input server address was not a valid TCP url. "
+                                   "It should match the form: " +
+                                   TCP_URL_REGEX_STR() );
+    }
+    m_zmq_socket.connect( m_server_address );
+
+    RCLCPP_INFO( this->get_logger(), "Creating subscriber" );
+    this->m_sub_updates = this->create_subscription< BBNUpdate >(
+      topic_update_msg, 1,
+      std::bind( &ZmqIntegrationClient::handle_bbn_update, this, _1 )
+      );
+  }
+
+  /// Destructor
+  ~ZmqIntegrationClient() override = default;
+
+  /**
+   * Handle a received update message, forwarding it over ZMQ to the configured
+   * server address.
+   *
+   * @param bbn_update_msg Update message to translate and forward.
+   */
+  void
+  handle_bbn_update( BBNUpdate::SharedPtr const bbn_update_msg )
+  {
+    auto const& log = this->get_logger();
+    RCLCPP_INFO( log, "Received ROS update message, translating "
+                      "into YAML and sending to ZMQ socket." );
+
+    YAML::Node yaml_root;
+    try
+    {
+      yaml_root = bbn_update_to_yaml( bbn_update_msg );
+    }
+    catch ( std::invalid_argument const& ex )
+    {
+      std::stringstream ss;
+      ss        << "Failed to translate input BBNUpdate message into yaml: "
+                << ex.what();
+      RCLCPP_ERROR( log, ss.str() );
+      return;
+    }
+
+    // Override the transmission timestamp to "now"
+    auto const now = std::chrono::system_clock::now();
+    auto epoch_seconds =
+      std::chrono::duration< double >( now.time_since_epoch() ).count();
+    yaml_root[ "header" ][ "transmit timestamp" ] = epoch_seconds;
+
+    auto yaml_translation = YAML::Dump( yaml_root );
+
+    // Should return the number of bytes sent. If return does not have a value,
+    // then the zmq_send function returned code EAGAIN (11) "Try again".
```

Comment:
Should we try again here or just log the warning and continue?

Reply:
I'm not really sure. I think this should be a discussion with Brian V. and the BBN folks on what the "failure" plan is. I may need to read some more ZMQ docs. I am currently intentionally dropping the message on the floor and continuing instead of being "stuck" trying to send the same message (causing backup on the ROS side, which *will* drop messages).

Reply:
I'm also not sure that EAGAIN "errors" are likely in our current client/server context: Neither the server nor the client (our thing) are using the `zmq::send_flags::dontwait` nor `zmq::recv_flags::dontwait` flags, which I think have to do with EAGAIN propagation.

Reply:
Yeah, it makes sense to not hold things up in the ROS side and get stuck in the callback function. That sounds good to me then.